src/test/java/com/impactupgrade/nucleus/it/MailchimpCommunicationServiceIT.java

```diff
@@ -0,0 +1,191 @@
+package com.impactupgrade.nucleus.it;
+
+import com.ecwid.maleorang.MailchimpObject;
+import com.ecwid.maleorang.method.v3_0.lists.members.MemberInfo;
+import com.impactupgrade.nucleus.App;
+import com.impactupgrade.nucleus.client.MailchimpClient;
+import com.impactupgrade.nucleus.environment.EnvironmentConfig;
+import com.impactupgrade.nucleus.model.CrmContact;
+import com.impactupgrade.nucleus.service.segment.CommunicationService;
+import com.impactupgrade.nucleus.service.segment.MailchimpCommunicationService;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MailchimpCommunicationServiceIT extends AbstractIT {
+
+  protected MailchimpCommunicationServiceIT() {
+    super(new App(envFactorySfdcMailchimp));
+  }
+
+  @Test
+  public void syncContacts() throws Exception {
+    List<String> emails = new ArrayList<>();
+    List<Object> values = new LinkedList<>();
+    values.add(List.of("Contact First Name", "Contact Last Name", "Contact Email", "Contact Email Opt In"));
+
+    for (int i = 1; i <= 5; i++) {
+      String firstname = RandomStringUtils.randomAlphabetic(8);
+      String lastname = RandomStringUtils.randomAlphabetic(8);
+      String email = (firstname + "." + lastname + "@gmail.com").toLowerCase(Locale.ROOT);
+
+      emails.add(email);
+      values.add(List.of(firstname, lastname, email, "true"));
+    }
+
+    // Bulk import to SF
+    Instant instant = LocalDate.now().atStartOfDay(ZoneId.of("UTC")).toInstant();
+    Calendar beforeBulkImport = Calendar.getInstance();
+    beforeBulkImport.setTimeInMillis(instant.toEpochMilli());
```

Comment:
Clever! I was wondering how we were going to prevent other contacts from getting in...